src/vs/workbench/contrib/terminal/browser/terminalQuickFixService.ts

```diff
@@ -35,16 +38,29 @@ export class TerminalQuickFixService implements ITerminalQuickFixService {
 	}
 
 	registerQuickFixProvider(id: string, provider: ITerminalQuickFixProvider): IDisposable {
-		this._providers.set(id, provider);
-		const selector = this._selectors.get(id);
-		if (!selector) {
-			throw new Error(`No registered selector for ID: ${id}`);
-		}
-		this._onDidRegisterProvider.fire({ selector, provider });
+		// This is more complicated than it looks like it should be because we need to return an
+		// IDisposable synchronously but we must await ITerminalContributionService.quickFixes
+		// asynchronously before actually registering the provider.
+		let disposed = false;
+		this._terminalContributionService.quickFixes.then(() => {
+			if (disposed) {
+				return;
+			}
+			this._providers.set(id, provider);
+			const selector = this._selectors.get(id);
+			if (!selector) {
```

Comment:
isn't this doing the same thing as 

https://github.com/microsoft/vscode/blob/6d72f80a3760290adfd34ba22ceff74cf096e1d4/src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts#L96-L100